evm/src/cpu/kernel/asm/core/precompiles/expmod.asm

```diff
@@ -1,3 +1,376 @@
+    %store_limbs
+    // stack: len, l_M, l_E, l_B, kexit_info
+
+    SWAP3
+    %pop3
+    // stack: len, kexit_info
+
+    PUSH expmod_contd
+    // stack: expmod_contd, len, kexit_info
+    DUP2
+    // stack: len, expmod_contd, len, kexit_info
+
+    DUP1
+    %mul_const(11)
+    // stack: s5=11*len, len, expmod_contd, len, kexit_info
+    SWAP1
+    // stack: len, s5, expmod_contd, len, kexit_info
+
+    DUP1
+    %mul_const(9)
+    // stack: s4=9*len, len, s5, expmod_contd, len, kexit_info
+    SWAP1
+    // stack: len, s4, s5, expmod_contd, len, kexit_info
+
+    DUP1
+    %mul_const(7)
+    // stack: s3=7*len, len, s4, s5, expmod_contd, len, kexit_info
+    SWAP1
+    // stack: len, s3, s4, s5, expmod_contd, len, kexit_info
+
+    DUP1
+    %mul_const(5)
+    // stack: s2=5*len, len, s3, s4, s5, expmod_contd, len, kexit_info
+    SWAP1
+    // stack: len, s2, s3, s4, s5, expmod_contd, len, kexit_info
+
+    DUP1
+    %mul_const(4)
+    // stack: s1=4*len, len, s2, s3, s4, s5, expmod_contd, len, kexit_info
+    SWAP1
+    // stack: len, s1, s2, s3, s4, s5, expmod_contd, len, kexit_info
+
+    DUP1
+    %mul_const(3)
+    // stack: out=3*len, len, s1, s2, s3, s4, s5, expmod_contd, len, kexit_info
+    SWAP1
+    // stack: len, out, s1, s2, s3, s4, s5, expmod_contd, len, kexit_info
+
+    DUP1
+    %mul_const(2)
+    // stack: m_loc=2*len, len, out, s1, s2, s3, s4, s5, expmod_contd, len, kexit_info
+    SWAP1
+    // stack: len, m_loc, out, s1, s2, s3, s4, s5, expmod_contd, len, kexit_info
+
+    PUSH 0
+    // stack: b_loc=0, e_loc=len, m_loc, out, s1, s2, s3, s4, s5, expmod_contd, len, kexit_info
+    DUP2
+    // stack: len, b_loc, e_loc, m_loc, out, s1, s2, s3, s4, s5, expmod_contd, len, kexit_info
+
+    %jump(modexp_bignum)
+
+expmod_contd:
+    // stack: len, kexit_info
+
+    // Copy the result value from kernel general memory to the parent's return data.
+
+    DUP1
+    // stack: len, len, kexit_info
+    %mstore_parent_context_metadata(@CTX_METADATA_RETURNDATA_SIZE)
+    // stack: len, kexit_info
+    DUP1
+    // stack: len, len, kexit_info
+    %mul_const(3)
+    // stack: out=3*len, len, kexit_info
+    PUSH 0
+    // stack: i=0, cur_address=out, len, kexit_info
+expmod_store_loop:
+    // stack: i, cur_address, len, kexit_info
+    DUP2
+    // stack: cur_address, i, cur_address, len, kexit_info
+    %mload_kernel_general
+    // stack: cur_limb, i, cur_address, len, kexit_info
+    DUP2
+    // stack: i, cur_limb, i, cur_address, len, kexit_info
+    %mul_const(16)
+    // stack: offset=16*i, cur_limb, i, cur_address, len, kexit_info
+    %stack (offset, cur_limb) -> (@SEGMENT_RETURNDATA, offset, cur_limb, 16)
+    // stack: @SEGMENT_RETURNDATA, offset, cur_limb, 16, i, cur_address, len, kexit_info
+    %mload_context_metadata(@CTX_METADATA_PARENT_CONTEXT)
+    // stack: parent_ctx, @SEGMENT_RETURNDATA, offset, cur_limb, 16, i, cur_address, len, kexit_info
+    %mstore_unpacking
```

Comment:
`mstore_unpacking` puts a scalar `offset'` on top of the stack which I think you don't account for here.